docs/source/daml/intro/2_Scenario.rst

```diff
@@ -0,0 +1,146 @@
+
+In the DAML IDE, you should see a small text reading "Scenario results" just above the line ``token_test_1 = do``. Click on it to display the outcome of the scenario.
+
+.. figure:: images/2_Scenario/scenario_results1.png
+
+This will open the scenario view in a separate column in VS Code. The default view is a tabular representation of the final state of the ledger:
+
+.. figure:: images/2_Scenario/tabular_view1.png
+
+The big title reading ``Intro_2_Scenario:Token`` is the identifier of the type of contract that's listed below. ``Intro_2_Scenario`` is the module name, ``Token`` the template name.
+
+The first columns, labelled vertically show which parties know about which contracts. In this simple scenario, the sole party "Alice" knows about the contract she created.
+
+The second column shows the ID of the contract. This will be explained later.
+
+The third column shows the status of the contract, either ``active`` or ``archived``. The remaining columns show the contract arguments, with one column per field. As expected, field ``owner`` is ``'Alice'``. The single quotation marks indicate that ``Alice`` is a party.
+
+To run the same test from the command line, save your module in a file ``Intro_2_Scenario.daml`` and run ``daml damlc -- test Intro_2_Scenario.daml``. If your file contains more than one scenario, all of them will be run.
+
+.. _intro_2_failure:
+
+Testing for Failure
+-------------------
+
+In :doc:`1_Token` you learned that creating a ``Token`` requires the authority of its owner. In other words, it should not be possible for Alice to create a Token for another party and vice versa. A reasonable attempt to test that would be:
+
+.. literalinclude:: daml/Intro_2_Scenario.daml
+  :language: daml
+  :start-after: -- FAILURE_TEST_1_BEGIN
+  :end-before: -- FAILURE_TEST_1_END
+
+However, if you open the scenario view for that scenario you, are faced with the following, message:
+
+.. figure:: images/2_Scenario/failure.png
+
+The scenario failed, as expected, but it aborted at the first failure so it only tested that Alice can't create a token for Bob. The second ``submit`` statement was never reached. To test for failing submits and keep the scenario running thereafter, or fail if the submission succeeds, DAML has the ``submitMustFail`` function:
+
+.. literalinclude:: daml/Intro_2_Scenario.daml
+  :language: daml
+  :start-after: -- TOKEN_TEST_2_BEGIN
+  :end-before: -- TOKEN_TEST_2_END
+
+``submitMustFail`` never has an impact on the ledger so the resulting tabular scenario view just shows the two Tokens resulting from the successful ``submit`` statements. Note the new column for Bob as well as the visibilities. Alice and Bob cannot see each others' Tokens.
+
+.. _archiving:
+
+Archiving Contracts
+-------------------
+
+Archival of contracts works just like creation, but using ``archive`` instead of ``create``. Whilst ``create`` takes an instance of a template, ``archive`` takes a reference to a contract.
```

Comment:
Prefer verb forms to nouns where possible - easier to parse

```suggestion
Archiving contracts works just like creating them, but using ``archive`` instead of ``create``. Where ``create`` takes an instance of a template, ``archive`` takes a reference to a contract.
```